### PR TITLE
feat(ui): new searchbox

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

**Comment by yesnault** [comment]
Seen recently (nothing searched)
Capture d’écran 2019-09-24 à 22 50 45

search with a project key or title (end with '/')
Capture d’écran 2019-09-24 à 22 50 50

classic search
Capture d’écran 2019-09-24 à 22 51 05

this will fix issue with sui-search component (not show the search results if user writes too fast the search)

**Comment by yesnault** [comment]
favorite stars everywhere on screenshot are fixed

File: ui/src/app/views/navbar/navbar.component.ts [hunk 1]
```diff
@@ -114,6 +125,22 @@ export class NavbarComponent implements OnInit, AfterViewInit {
             }
         });
 
+        // Listen change on recent projects viewed
+        this._projectStore.getRecentProjects().subscribe(projects => {
```
Comment (sguiheux) [inline ui/src/app/views/navbar/navbar.component.ts]: Missing subscription
Check if there is AutoUnsubscribe on component

File: ui/src/app/views/navbar/navbar.component.ts [hunk 2]
```diff
@@ -114,6 +125,22 @@ export class NavbarComponent implements OnInit, AfterViewInit {
...
+                })).concat(
+                    this.recentItems.filter((i) => i.type !== 'project')
+                );
+                this.items = this.recentItems;
+                this._cd.detectChanges();
+            }
+        });
+
         // Listen change on recent app viewed
         this._appStore.getRecentApplications().subscribe(apps => {
```
Comment (sguiheux) [inline ui/src/app/views/navbar/navbar.component.ts]: Missing subscription

File: ui/src/app/views/navbar/navbar.component.ts [hunk 3]
```diff
@@ -114,6 +125,22 @@ export class NavbarComponent implements OnInit, AfterViewInit {
...
+                this.recentItems = projects.toArray().map((prj) => ({
+                    type: 'project',
+                    value: prj.project_key,
+                    title: prj.name,
+                    projectKey: prj.project_key
+                })).concat(
+                    this.recentItems.filter((i) => i.type !== 'project')
+                );
+                this.items = this.recentItems;
+                this._cd.detectChanges();
```
Comment (sguiheux) [inline ui/src/app/views/navbar/navbar.component.ts]: Try with . markForCheck()

File: ui/src/app/views/navbar/navbar.component.ts (hunk 1 above)
Comment (yesnault) [reply ui/src/app/views/navbar/navbar.component.ts]: done

File: ui/src/app/views/navbar/navbar.component.ts (hunk 3 above)
Comment (yesnault) [reply ui/src/app/views/navbar/navbar.component.ts]: done

File: ui/src/app/views/navbar/navbar.component.ts (hunk 2 above)
Comment (yesnault) [reply ui/src/app/views/navbar/navbar.component.ts]: done

**Pull request merged**